src/loaders/markdown/transformer/rehypeHighlightLine.ts

```diff
@@ -1,69 +1,45 @@
 import type { Root } from 'hast';
 import type { Transformer } from 'unified';
-import type { IMdTransformerOptions } from '.';
 
 let visit: typeof import('unist-util-visit').visit;
 let isElement: typeof import('hast-util-is-element').isElement;
 
-const RE = /{([\d,-]+)}/;
+const RE = /{((?:\d+(?:-\d+)?,?)+)}/;
 
 // workaround to import pure esm module
 (async () => {
   ({ visit } = await import('unist-util-visit'));
   ({ isElement } = await import('hast-util-is-element'));
 })();
 
-type IRehypeHighlightLine = Pick<IMdTransformerOptions, 'resolve'>;
-
-const attrsToLines = (rawAttrs: string) => {
-  const attrs = rawAttrs.replace(/^(?:\[.*?\])?.*?([\d,-]+).*/, '$1').trim();
+const attrsToLines = (attrs: string) => {
   const result: number[] = [];
   attrs
     ?.split(',')
     .map((v) => v.split('-').map((v) => parseInt(v, 10)))
     .forEach(([start, end]) => {
```

Comment:
```suggestion
    .forEach(([start, end = start]) => {
```
然后就可以直接用 `for` 不需要 `if` 判断了？